AK/StdLibExtras.h

```diff
@@ -65,6 +65,17 @@ constexpr T&& move(T& arg)
     return static_cast<T&&>(arg);
 }
 
+#ifndef NO_STRICT_MOVE
```

Comment:
`NO_STRICT_MOVE` implies the existence of something like a `STRICT_MOVE` definition. Maybe call this `ALLOW_CONST_REFERENCE_MOVE` instead?